frontend/src/authentication/components/AuthGuard.js

```diff
@@ -23,6 +31,22 @@ export const AuthGuard = (props) => {
     return <Navigate to={requestedLocation} />;
   }
 
+  if (
+    sessionStorage.getItem('window-location') &&
+    location.pathname !== sessionStorage.getItem('window-location')
+  ) {
+    const windowPathLocation = sessionStorage.getItem('window-location');
+    sessionStorage.removeItem('window-location');
+    // Check if the window-location only contains alphanumeric and / in it and its not tampered
+    if (!regexToValidateWindowPathName.test(windowPathLocation))
+      return <>{children}</>;
+    // A guardrail to limit the string of the pathname to a certain characters
+    if (windowPathLocation.length > 50) return <>{children}</>;
```

Comment:
What's wrong with long URLs? Why 50? Maybe it's better to make it a param, not magic constant.

Reply:
Hi @SofiaSazonova , Thanks for the review. I added this number by heuristically checking the length of the windowpath for the longest URL. I think it was the URL to table in a dataset and it was approx 33 - 37 characters. 

I can make it a param and store it in the contants.js . What are your thoughts ?

Reply:
I'd rather make it a setting, rather then a constant. 
The maximum URL length is something the customer may want to define before the deployment, and it's more an attribute of solution rather then code constant.
But maybe @dlpzx can correct me here.

Reply:
@TejasRGitHub in this case, `constants` is a good place for it.

Reply:
Hi @SofiaSazonova , Updated code to make it a constant . Please let me know if it looks good